docs/tutorial/code/flask/task.yaml

```diff
@@ -0,0 +1,212 @@
+  rockcraft.skopeo --insecure-policy copy --dest-tls-verify=false \
+    oci-archive:flask-hello-world_0.1_$(dpkg --print-architecture).rock \
+    docker://localhost:32000/flask-hello-world:0.1
+  # [docs:skopeo-copy-end]
+
+  # [docs:create-charm-dir]
+  mkdir charm
+  cd charm
+  # [docs:create-charm-dir-end]
+
+  # [docs:charm-init]
+  charmcraft init --profile flask-framework --name flask-hello-world
+  # [docs:charm-init-end]
+
+  # [docs:charm-pack]
+  charmcraft pack
+  # [docs:charm-pack-end]
+
+  # [docs:ls-charm]
+  ls *.charm -l
+  # [docs:ls-charm-end]
+
+  # [docs:add-juju-model]
+  juju add-model flask-hello-world
+  # [docs:add-juju-model-end]
+
+  juju set-model-constraints -m flask-hello-world arch=$(dpkg --print-architecture)
+
+  # [docs:deploy-juju-model]
+  juju deploy ./flask-hello-world_ubuntu-22.04-$(dpkg --print-architecture).charm \
+    flask-hello-world --resource \
+    flask-app-image=localhost:32000/flask-hello-world:0.1
+  # [docs:deploy-juju-model-end]
+
+  # [docs:deploy-nginx]
+  juju deploy nginx-ingress-integrator --channel=latest/edge --revision 122
+  juju integrate nginx-ingress-integrator flask-hello-world
+  # [docs:deploy-nginx-end]
+
+  # [docs:config-nginx]
+  juju config nginx-ingress-integrator \
+    service-hostname=flask-hello-world path-routes=/
+  # [docs:config-nginx-end]
+
+  # give Juju some time to deploy the apps
+  juju wait-for application flask-hello-world --query='status=="active"' --timeout 10m | juju status
+  juju wait-for application nginx-ingress-integrator --query='status=="active"' --timeout 10m | juju status
+
+  # [docs:curl-init-deployment]
+  curl http://flask-hello-world --resolve flask-hello-world:80:127.0.0.1
+  # [docs:curl-init-deployment-end]
+
+  juju status
+
+  cd ..
+  cat greeting_app.py > app.py
+  sed -i "s/version: .*/version: 0.2/g" rockcraft.yaml
+  rockcraft pack
+
+  # [docs:docker-update]
+  rockcraft.skopeo --insecure-policy copy --dest-tls-verify=false \
+    oci-archive:flask-hello-world_0.2_$(dpkg --print-architecture).rock \
+    docker://localhost:32000/flask-hello-world:0.2
+  # [docs:docker-update-end]
+
+  cat greeting_charmcraft.yaml >> ./charm/charmcraft.yaml
+  cd charm
+  charmcraft pack
+
+  # [docs:refresh-deployment]
+  juju refresh flask-hello-world \
+    --path=./flask-hello-world_ubuntu-22.04-$(dpkg --print-architecture).charm \
+    --resource flask-app-image=localhost:32000/flask-hello-world:0.2
+  # [docs:refresh-deployment-end]
+
+  # give Juju some time to refresh the app
+  juju wait-for application flask-hello-world --query='status=="active"' --timeout 10m | juju status
+
+  curl http://flask-hello-world --resolve flask-hello-world:80:127.0.0.1
+
+  # [docs:change-config]
+  juju config flask-hello-world greeting='Hi!'
+  # [docs:change-config-end]
+
+  juju wait-for application flask-hello-world --query='status=="active"' --timeout 10m | juju status
+  curl http://flask-hello-world --resolve flask-hello-world:80:127.0.0.1
```

Comment:
Perhaps we check that the response is `hi` here, e.g., `curl http://flask-hello-world --resolve flask-hello-world:80:127.0.0.1 | grep hi`

Reply:
perhaps unclude a comment for the reason for grep as well